connector/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -43,7 +43,11 @@ message Expression {
     Expression expr = 1;
 
     // (Required) the data type that the expr to be casted to.
-    DataType cast_to_type = 2;
+    oneof cast_to_type {
```

Comment:
FYI: this is a breaking change in the proto message. Ideally, we would use

```
reserved 2;
oneof cast_to_type {
  DataType type = 3;
  string type_str = 4;
}
```

Reply:
Ah thanks. I didn't know this way to evolve the proto.

Given Spark Connect is still alpha component though for now we don't need to be enforced to maintain the backwards compatibility

Reply:
I think it's fine since this message has not been actually used?

Reply:
We're ok now. We can use Buf to check for breaking changes.